display/src/lib.rs

```diff
@@ -114,22 +120,28 @@ impl<N: Network> Display<N> {
 
 impl<N: Network> Display<N> {
     /// Initializes the logger.
-    fn initialize_logger(verbosity: u8, nodisplay: bool) -> mpsc::Receiver<Vec<u8>> {
+    fn initialize_logger<P: AsRef<Path>>(verbosity: u8, nodisplay: bool, logfile: P) -> mpsc::Receiver<Vec<u8>> {
         match verbosity {
             0 => std::env::set_var("RUST_LOG", "info"),
             1 => std::env::set_var("RUST_LOG", "debug"),
             2 | 3 => std::env::set_var("RUST_LOG", "trace"),
             _ => std::env::set_var("RUST_LOG", "info"),
         };
 
-        // Filter out undesirable logs.
-        let filter = EnvFilter::from_default_env()
-            .add_directive("mio=off".parse().unwrap())
-            .add_directive("tokio_util=off".parse().unwrap())
-            .add_directive("hyper=off".parse().unwrap())
-            .add_directive("reqwest=off".parse().unwrap())
-            .add_directive("want=off".parse().unwrap())
-            .add_directive("warp=off".parse().unwrap());
+        // Filter out undesirable logs. (unfortunately EnvFilter cannot be cloned)
+        let [filter, filter2] = std::array::from_fn(|_| {
+            EnvFilter::from_default_env()
+                .add_directive("mio=off".parse().unwrap())
+                .add_directive("tokio_util=off".parse().unwrap())
+                .add_directive("hyper=off".parse().unwrap())
+                .add_directive("reqwest=off".parse().unwrap())
+                .add_directive("want=off".parse().unwrap())
+                .add_directive("warp=off".parse().unwrap())
+        });
+
+        // Create a file to write logs to.
+        let logfile =
+            File::options().append(true).create(true).open(logfile).expect("Failed to open the file for writing logs");
```

Comment:
is panic there ok?

Reply:
I'd say so, the possible errors are permissions with those options set.